feat: add submit and wait for transaction submission (#528)
XRPLF/xrpl-py: CHANGELOG.md

and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [[Unreleased]]
### Added:
- Added `submit_and_wait` to sign (if needed), autofill, submit a transaction and wait for its final outcome

> - `submit` and `send_reliable_submission` now accepting a boolean param `fail_hard` (default to `False`; if `True`, halt the submission

Reviewer: nit:
```suggestion
- `submit` and `send_reliable_submission` now accept an optional boolean param `fail_hard` (if `True` halt the submission if it's not immediately validated)
```

> - `submit` and `send_reliable_submission` now accepting a boolean param `fail_hard` (default to `False`; if `True`, halt the submission
> if fails locally)

Reviewer: I think this wording is a bit more clear - although maybe I'm misunderstanding what "fails locally" means?

```suggestion
- `submit` and `send_reliable_submission` now accept an optional boolean param `fail_hard` (if `True` halt the submission if it's not immediately validated)
```

- Added `getNFTokenID` to get the NFTokenID after minting a token.
- Added `LedgerEntryType` enum and added `type` field to `Ledger` and `LedgerData` requests
- Added the algorithm used to encode a wallet's seed to the wallet.

> - Added `submit_and_wait` to sign (if needed), autofill, submit a transaction and wait for its final outcome

Reviewer 2: ```suggestion
- Added `submit_and_wait` to autofill, sign, and submit a transaction and wait for its final outcome
```

Reviewer 2: The changelog should also have a note about `send_reliable_submission` now accepting a `fail_hard` param

Reviewer 2: Also, this needs to move to the `Unreleased` section
